src/utils/multicall.ts

```diff
@@ -11,15 +11,46 @@ interface Call {
 }
 
 const multicall = async (abi: any[], calls: Call[]) => {
-  const web3 = getWeb3NoAccount()
-  const multi = new web3.eth.Contract(MultiCallAbi as unknown as AbiItem, getMulticallAddress())
-  const itf = new Interface(abi)
+  try {
+    const web3 = getWeb3NoAccount()
+    const multi = new web3.eth.Contract(MultiCallAbi as unknown as AbiItem, getMulticallAddress())
+    const itf = new Interface(abi)
 
-  const calldata = calls.map((call) => [call.address.toLowerCase(), itf.encodeFunctionData(call.name, call.params)])
-  const { returnData } = await multi.methods.aggregate(calldata).call()
-  const res = returnData.map((call, i) => itf.decodeFunctionResult(calls[i].name, call))
+    const calldata = calls.map((call) => [call.address.toLowerCase(), itf.encodeFunctionData(call.name, call.params)])
+    const { returnData } = await multi.methods.aggregate(calldata).call()
+    const res = returnData.map((call, i) => itf.decodeFunctionResult(calls[i].name, call))
 
-  return res
+    return res
+  } catch (error) {
+    throw new Error(error)
```

Comment:
Throwing the error here results in better console messages.

Reply:
there is no reason to try catch if you re throw the error.